packages/backend/src/problems/problem.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+
+import { CreateProblemDto } from "./create-problem.dto";
+import { Problem } from "./problem.model";
+
+@Injectable()
+export class ProblemService {
+    constructor(@InjectRepository(Problem) private readonly problemRepo: Repository<Problem>) {}
+
+    async createProblem(dto: CreateProblemDto): Promise<Problem> {
```

Comment:
maybe we can just call it `create`? then it reads as `problemService.create(problemInfo)`.